dcos/dcos-ui
# [spike] DCOS-20182: Error Validation in Delete Modal

Show an error to the user when something goes wrong
with deleting the service when we don't have details.

Closes https://jira.mesosphere.com/browse/DCOS-20182

## Testing
0. In `dcos-ui/plugins/services/src/js/events/MarathonActions.js` in the `deleteService` method above the `RequestUtil`, add something like `url = "123"`, in order to show an error.
Similarity, in `dcos/dcos-ui/src/js/events/CosmosPackagesActions.js` in the `uninstallPackage` method, hard-code packageName to something like "123".
1. Start a service.
2. Try to delete it.
3. Verify that an error is shown instead of nothing happening.
4. Close the modal and verify that a toast notification is shown.
5. Try the same for pods.
6. Try the same for catalog packages.
7. (Optional) Add some fake translations in the Chinese message catalog and verify that they work.

Thanks @TattdCodeMonkey for explaining to me the way toast works.

## Trade-offs
I am not sure how to create a real error, so I am mocking one and in this case it looks like an error won't display because no details are passed. So I decided to create a default error message, which could have some room for improvement. @mperrotti , what do you think?

The toast capitalizes the service name.

## Dependencies
None.

## Screenshots
![Peek 2019-07-19 17-18](https://user-images.githubusercontent.com/40791275/61542002-5eb89380-aa49-11e9-8f48-d16c2b3c4923.gif)

![Peek 2019-07-19 17-18-2](https://user-images.githubusercontent.com/40791275/61542010-637d4780-aa49-11e9-89df-98232c85cc7f.gif)
